core/src/main/java/ai/timefold/solver/core/api/score/analysis/ConstraintAnalysis.java

```diff
@@ -200,6 +200,28 @@ public String constraintPackage() {
      */
     @SuppressWarnings("java:S3457")
     public @NonNull String summarize() {
+        return buildSummary(DEFAULT_SUMMARY_CONSTRAINT_MATCH_LIMIT);
+    }
+
+    /**
+     * Return a diagnostic text that explains part of the score quality using the {@link #summarize(int)} method
+     * in a similar way to {@link #summarize()}.
+     * It is possible to specify the maximum number of constraint matches to display per constraint by passing
+     * the desired limit as an argument.
+     *
+     * @param topLimit maximum number of constraint matches to display per constraint
+     *        Use {@link Integer#MAX_VALUE} to show all matches.
+     */
+
+    public @NonNull String summarize(int topLimit) {
+        if (topLimit < 1) {
+            throw new IllegalArgumentException("The topLimit (" + topLimit + ") must be at least 1.");
```

Comment:
We now use a new convention for exception messages with arguments.

    throw new IllegalArgumentException("The topLimit (%d) must be at least 1.".formatted(topLimit));

Provides better readability for long exception messages.
(Not so much for these short ones, but conventions are conventions.)